AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -5024,7 +5088,7 @@ function Join-AzStorageAccount {
                     -ErrorAction Stop
             }
             
-            Assert-IsNativeAD -StorageAccount $StorageAccount
+            Assert-IsUnconfiguredOrNativeAD -StorageAccount $StorageAccount

```

Comment:
Should this actually allow the value `"AD"` or not?

Reply:
I think we should allow the value "AD". If users want to re-join, then on Azure Portal the setting might already been set to "AD" in the previous run of Join-AzStorageAccount. Not allowing it could prevent users from re-join.

Reply:
Worked on this some more today, and came to the same conclusion! I fully agree, let's keep it as is.